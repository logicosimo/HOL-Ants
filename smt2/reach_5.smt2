(declare-datatype Position ((P0) (P1) (P2) (P3) (P4)))
(declare-const a Int)
(declare-const a_1 Int)
(declare-const a_2 Int)
(declare-const a_3 Int)
(declare-const a_4 Int)
(declare-const a_5 Int)
(declare-const a_6 Int)
(declare-const a_7 Int)
(declare-const a0 Position)
(declare-const a1 Position)
(declare-const a2 Position)
(declare-const a3 Position)
(declare-const a4 Position)
(declare-const c Int)
(declare-const c_1 Int)
(declare-const c_2 Int)
(declare-const c_3 Int)
(declare-const d0 Bool)
(declare-const d1 Bool)
(declare-const d2 Bool)
(declare-const d3 Bool)
(declare-const d4 Bool)
(declare-const dir Bool)
(declare-const dir_1 Bool)
(declare-const dir_2 Bool)
(declare-const dir_3 Bool)
(declare-const dir_4 Bool)
(declare-const dir_5 Bool)
(declare-const dir_6 Bool)
(declare-const dir_7 Bool)
(declare-const dir_8 Bool)
(declare-const dir_9 Bool)
(declare-const dir_10 Bool)
(declare-const dir_11 Bool)
(declare-const dir_12 Bool)
(declare-const dir_13 Bool)
(declare-const dir_14 Bool)
(declare-const dir_15 Bool)
(declare-const dir_16 Bool)
(declare-const dir_17 Bool)
(declare-const dir_18 Bool)
(declare-const dir_19 Bool)
(declare-const pos Position)
(declare-const pos_1 Position)
(declare-const pos_2 Position)
(declare-const pos_3 Position)
(declare-const pos_4 Position)
(declare-const pos_5 Position)
(declare-const pos_6 Position)
(declare-const pos_7 Position)
(declare-const pos_8 Position)
(declare-const pos_9 Position)
(declare-const pos_10 Position)
(declare-const pos_11 Position)
(declare-const pos_12 Position)
(declare-const pos_13 Position)
(declare-const pos_14 Position)
(declare-const pos_15 Position)
(declare-const pos_16 Position)
(declare-const pos_17 Position)
(declare-const pos_18 Position)
(declare-const pos_19 Position)
(declare-const s1 Int)
(declare-const s2 Int)
(declare-const s3 Int)
(assert (>= a 0))
(assert (>= a_1 0))
(assert (>= a_2 0))
(assert (>= a_3 0))
(assert (>= a_4 0))
(assert (>= a_5 0))
(assert (>= a_6 0))
(assert (>= a_7 0))
(assert (>= c 0))
(assert (>= c_1 0))
(assert (>= c_2 0))
(assert (>= c_3 0))
(assert (>= s1 0))
(assert (>= s2 0))
(assert (>= s3 0))
(assert
 (and (= a_6 0) (= c_3 0) (= a_7 0) (or (= pos_15 P1) (= pos_15 P2)) dir_15
  (or (= pos_19 P1) (= pos_19 P2)) dir_19 (or (= pos_16 P1) (= pos_16 P2))
  dir_16 (or (= pos_17 P1) (= pos_17 P2)) dir_17
  (or (= pos_18 P1) (= pos_18 P2)) dir_18
  (= a_4
   (+ a_6 (ite (= pos_15 P1) 1 0) (ite (= pos_17 P1) 1 0)
    (ite (= pos_16 P1) 1 0) (ite (= pos_18 P1) 1 0) (ite (= pos_19 P1) 1 0)))
  (= c_2
   (+ c_3 (ite (= pos_15 P3) 1 0) (ite (= pos_17 P3) 1 0)
    (ite (= pos_16 P3) 1 0) (ite (= pos_18 P3) 1 0) (ite (= pos_19 P3) 1 0)))
  (= a_5
   (+ a_7 (ite (= pos_15 P2) 1 0) (ite (= pos_17 P2) 1 0)
    (ite (= pos_16 P2) 1 0) (ite (= pos_18 P2) 1 0) (ite (= pos_19 P2) 1 0)))
  (=> (= pos_15 P1) (and (= pos_10 (ite dir_15 P4 P0)) (iff dir_10 dir_15)))
  (=> (= pos_15 P2) (and (= pos_10 (ite dir_15 P3 P0)) (iff dir_10 dir_15)))
  (=> (= pos_15 P3) (and (= pos_10 (ite dir_15 P4 P2)) (iff dir_10 dir_15)))
  (=> (= pos_15 P0) (and (or (= pos_10 P1) (= pos_10 P2)) dir_10))
  (=> (and (= pos_15 P0) (= pos_10 P1)) (<= a_7 a_6))
  (=> (and (= pos_15 P0) (= pos_10 P2)) (<= a_6 a_7))
  (=> (= pos_15 P4) (and (or (= pos_10 P1) (= pos_10 P3)) (not dir_10)))
  (=> (and (= pos_15 P4) (= pos_10 P1)) (<= c_3 a_6))
  (=> (and (= pos_15 P4) (= pos_10 P3)) (<= a_6 c_3))
  (=> (= pos_19 P1) (and (= pos_14 (ite dir_19 P4 P0)) (iff dir_14 dir_19)))
  (=> (= pos_19 P2) (and (= pos_14 (ite dir_19 P3 P0)) (iff dir_14 dir_19)))
  (=> (= pos_19 P3) (and (= pos_14 (ite dir_19 P4 P2)) (iff dir_14 dir_19)))
  (=> (= pos_19 P0) (and (or (= pos_14 P1) (= pos_14 P2)) dir_14))
  (=> (and (= pos_19 P0) (= pos_14 P1)) (<= a_7 a_6))
  (=> (and (= pos_19 P0) (= pos_14 P2)) (<= a_6 a_7))
  (=> (= pos_19 P4) (and (or (= pos_14 P1) (= pos_14 P3)) (not dir_14)))
  (=> (and (= pos_19 P4) (= pos_14 P1)) (<= c_3 a_6))
  (=> (and (= pos_19 P4) (= pos_14 P3)) (<= a_6 c_3))
  (=> (= pos_16 P1) (and (= pos_11 (ite dir_16 P4 P0)) (iff dir_11 dir_16)))
  (=> (= pos_16 P2) (and (= pos_11 (ite dir_16 P3 P0)) (iff dir_11 dir_16)))
  (=> (= pos_16 P3) (and (= pos_11 (ite dir_16 P4 P2)) (iff dir_11 dir_16)))
  (=> (= pos_16 P0) (and (or (= pos_11 P1) (= pos_11 P2)) dir_11))
  (=> (and (= pos_16 P0) (= pos_11 P1)) (<= a_7 a_6))
  (=> (and (= pos_16 P0) (= pos_11 P2)) (<= a_6 a_7))
  (=> (= pos_16 P4) (and (or (= pos_11 P1) (= pos_11 P3)) (not dir_11)))
  (=> (and (= pos_16 P4) (= pos_11 P1)) (<= c_3 a_6))
  (=> (and (= pos_16 P4) (= pos_11 P3)) (<= a_6 c_3))
  (=> (= pos_17 P1) (and (= pos_12 (ite dir_17 P4 P0)) (iff dir_12 dir_17)))
  (=> (= pos_17 P2) (and (= pos_12 (ite dir_17 P3 P0)) (iff dir_12 dir_17)))
  (=> (= pos_17 P3) (and (= pos_12 (ite dir_17 P4 P2)) (iff dir_12 dir_17)))
  (=> (= pos_17 P0) (and (or (= pos_12 P1) (= pos_12 P2)) dir_12))
  (=> (and (= pos_17 P0) (= pos_12 P1)) (<= a_7 a_6))
  (=> (and (= pos_17 P0) (= pos_12 P2)) (<= a_6 a_7))
  (=> (= pos_17 P4) (and (or (= pos_12 P1) (= pos_12 P3)) (not dir_12)))
  (=> (and (= pos_17 P4) (= pos_12 P1)) (<= c_3 a_6))
  (=> (and (= pos_17 P4) (= pos_12 P3)) (<= a_6 c_3))
  (=> (= pos_18 P1) (and (= pos_13 (ite dir_18 P4 P0)) (iff dir_13 dir_18)))
  (=> (= pos_18 P2) (and (= pos_13 (ite dir_18 P3 P0)) (iff dir_13 dir_18)))
  (=> (= pos_18 P3) (and (= pos_13 (ite dir_18 P4 P2)) (iff dir_13 dir_18)))
  (=> (= pos_18 P0) (and (or (= pos_13 P1) (= pos_13 P2)) dir_13))
  (=> (and (= pos_18 P0) (= pos_13 P1)) (<= a_7 a_6))
  (=> (and (= pos_18 P0) (= pos_13 P2)) (<= a_6 a_7))
  (=> (= pos_18 P4) (and (or (= pos_13 P1) (= pos_13 P3)) (not dir_13)))
  (=> (and (= pos_18 P4) (= pos_13 P1)) (<= c_3 a_6))
  (=> (and (= pos_18 P4) (= pos_13 P3)) (<= a_6 c_3))
  (= a_2
   (+ a_4 (ite (= pos_10 P1) 1 0) (ite (= pos_12 P1) 1 0)
    (ite (= pos_11 P1) 1 0) (ite (= pos_13 P1) 1 0) (ite (= pos_14 P1) 1 0)))
  (= c_1
   (+ c_2 (ite (= pos_10 P3) 1 0) (ite (= pos_12 P3) 1 0)
    (ite (= pos_11 P3) 1 0) (ite (= pos_13 P3) 1 0) (ite (= pos_14 P3) 1 0)))
  (= a_3
   (+ a_5 (ite (= pos_10 P2) 1 0) (ite (= pos_12 P2) 1 0)
    (ite (= pos_11 P2) 1 0) (ite (= pos_13 P2) 1 0) (ite (= pos_14 P2) 1 0)))
  (=> (= pos_10 P1) (and (= pos_5 (ite dir_10 P4 P0)) (iff dir_5 dir_10)))
  (=> (= pos_10 P2) (and (= pos_5 (ite dir_10 P3 P0)) (iff dir_5 dir_10)))
  (=> (= pos_10 P3) (and (= pos_5 (ite dir_10 P4 P2)) (iff dir_5 dir_10)))
  (=> (= pos_10 P0) (and (or (= pos_5 P1) (= pos_5 P2)) dir_5))
  (=> (and (= pos_10 P0) (= pos_5 P1)) (<= a_5 a_4))
  (=> (and (= pos_10 P0) (= pos_5 P2)) (<= a_4 a_5))
  (=> (= pos_10 P4) (and (or (= pos_5 P1) (= pos_5 P3)) (not dir_5)))
  (=> (and (= pos_10 P4) (= pos_5 P1)) (<= c_2 a_4))
  (=> (and (= pos_10 P4) (= pos_5 P3)) (<= a_4 c_2))
  (=> (= pos_14 P1) (and (= pos_9 (ite dir_14 P4 P0)) (iff dir_9 dir_14)))
  (=> (= pos_14 P2) (and (= pos_9 (ite dir_14 P3 P0)) (iff dir_9 dir_14)))
  (=> (= pos_14 P3) (and (= pos_9 (ite dir_14 P4 P2)) (iff dir_9 dir_14)))
  (=> (= pos_14 P0) (and (or (= pos_9 P1) (= pos_9 P2)) dir_9))
  (=> (and (= pos_14 P0) (= pos_9 P1)) (<= a_5 a_4))
  (=> (and (= pos_14 P0) (= pos_9 P2)) (<= a_4 a_5))
  (=> (= pos_14 P4) (and (or (= pos_9 P1) (= pos_9 P3)) (not dir_9)))
  (=> (and (= pos_14 P4) (= pos_9 P1)) (<= c_2 a_4))
  (=> (and (= pos_14 P4) (= pos_9 P3)) (<= a_4 c_2))
  (=> (= pos_11 P1) (and (= pos_6 (ite dir_11 P4 P0)) (iff dir_6 dir_11)))
  (=> (= pos_11 P2) (and (= pos_6 (ite dir_11 P3 P0)) (iff dir_6 dir_11)))
  (=> (= pos_11 P3) (and (= pos_6 (ite dir_11 P4 P2)) (iff dir_6 dir_11)))
  (=> (= pos_11 P0) (and (or (= pos_6 P1) (= pos_6 P2)) dir_6))
  (=> (and (= pos_11 P0) (= pos_6 P1)) (<= a_5 a_4))
  (=> (and (= pos_11 P0) (= pos_6 P2)) (<= a_4 a_5))
  (=> (= pos_11 P4) (and (or (= pos_6 P1) (= pos_6 P3)) (not dir_6)))
  (=> (and (= pos_11 P4) (= pos_6 P1)) (<= c_2 a_4))
  (=> (and (= pos_11 P4) (= pos_6 P3)) (<= a_4 c_2))
  (=> (= pos_12 P1) (and (= pos_7 (ite dir_12 P4 P0)) (iff dir_7 dir_12)))
  (=> (= pos_12 P2) (and (= pos_7 (ite dir_12 P3 P0)) (iff dir_7 dir_12)))
  (=> (= pos_12 P3) (and (= pos_7 (ite dir_12 P4 P2)) (iff dir_7 dir_12)))
  (=> (= pos_12 P0) (and (or (= pos_7 P1) (= pos_7 P2)) dir_7))
  (=> (and (= pos_12 P0) (= pos_7 P1)) (<= a_5 a_4))
  (=> (and (= pos_12 P0) (= pos_7 P2)) (<= a_4 a_5))
  (=> (= pos_12 P4) (and (or (= pos_7 P1) (= pos_7 P3)) (not dir_7)))
  (=> (and (= pos_12 P4) (= pos_7 P1)) (<= c_2 a_4))
  (=> (and (= pos_12 P4) (= pos_7 P3)) (<= a_4 c_2))
  (=> (= pos_13 P1) (and (= pos_8 (ite dir_13 P4 P0)) (iff dir_8 dir_13)))
  (=> (= pos_13 P2) (and (= pos_8 (ite dir_13 P3 P0)) (iff dir_8 dir_13)))
  (=> (= pos_13 P3) (and (= pos_8 (ite dir_13 P4 P2)) (iff dir_8 dir_13)))
  (=> (= pos_13 P0) (and (or (= pos_8 P1) (= pos_8 P2)) dir_8))
  (=> (and (= pos_13 P0) (= pos_8 P1)) (<= a_5 a_4))
  (=> (and (= pos_13 P0) (= pos_8 P2)) (<= a_4 a_5))
  (=> (= pos_13 P4) (and (or (= pos_8 P1) (= pos_8 P3)) (not dir_8)))
  (=> (and (= pos_13 P4) (= pos_8 P1)) (<= c_2 a_4))
  (=> (and (= pos_13 P4) (= pos_8 P3)) (<= a_4 c_2))
  (= a
   (+ a_2 (ite (= pos_5 P1) 1 0) (ite (= pos_7 P1) 1 0)
    (ite (= pos_6 P1) 1 0) (ite (= pos_8 P1) 1 0) (ite (= pos_9 P1) 1 0)))
  (= c
   (+ c_1 (ite (= pos_5 P3) 1 0) (ite (= pos_7 P3) 1 0)
    (ite (= pos_6 P3) 1 0) (ite (= pos_8 P3) 1 0) (ite (= pos_9 P3) 1 0)))
  (= a_1
   (+ a_3 (ite (= pos_5 P2) 1 0) (ite (= pos_7 P2) 1 0)
    (ite (= pos_6 P2) 1 0) (ite (= pos_8 P2) 1 0) (ite (= pos_9 P2) 1 0)))
  (=> (= pos_5 P1) (and (= pos (ite dir_5 P4 P0)) (iff dir dir_5)))
  (=> (= pos_5 P2) (and (= pos (ite dir_5 P3 P0)) (iff dir dir_5)))
  (=> (= pos_5 P3) (and (= pos (ite dir_5 P4 P2)) (iff dir dir_5)))
  (=> (= pos_5 P0) (and (or (= pos P1) (= pos P2)) dir))
  (=> (and (= pos_5 P0) (= pos P1)) (<= a_3 a_2))
  (=> (and (= pos_5 P0) (= pos P2)) (<= a_2 a_3))
  (=> (= pos_5 P4) (and (or (= pos P1) (= pos P3)) (not dir)))
  (=> (and (= pos_5 P4) (= pos P1)) (<= c_1 a_2))
  (=> (and (= pos_5 P4) (= pos P3)) (<= a_2 c_1))
  (=> (= pos_9 P1) (and (= pos_4 (ite dir_9 P4 P0)) (iff dir_4 dir_9)))
  (=> (= pos_9 P2) (and (= pos_4 (ite dir_9 P3 P0)) (iff dir_4 dir_9)))
  (=> (= pos_9 P3) (and (= pos_4 (ite dir_9 P4 P2)) (iff dir_4 dir_9)))
  (=> (= pos_9 P0) (and (or (= pos_4 P1) (= pos_4 P2)) dir_4))
  (=> (and (= pos_9 P0) (= pos_4 P1)) (<= a_3 a_2))
  (=> (and (= pos_9 P0) (= pos_4 P2)) (<= a_2 a_3))
  (=> (= pos_9 P4) (and (or (= pos_4 P1) (= pos_4 P3)) (not dir_4)))
  (=> (and (= pos_9 P4) (= pos_4 P1)) (<= c_1 a_2))
  (=> (and (= pos_9 P4) (= pos_4 P3)) (<= a_2 c_1))
  (=> (= pos_6 P1) (and (= pos_1 (ite dir_6 P4 P0)) (iff dir_1 dir_6)))
  (=> (= pos_6 P2) (and (= pos_1 (ite dir_6 P3 P0)) (iff dir_1 dir_6)))
  (=> (= pos_6 P3) (and (= pos_1 (ite dir_6 P4 P2)) (iff dir_1 dir_6)))
  (=> (= pos_6 P0) (and (or (= pos_1 P1) (= pos_1 P2)) dir_1))
  (=> (and (= pos_6 P0) (= pos_1 P1)) (<= a_3 a_2))
  (=> (and (= pos_6 P0) (= pos_1 P2)) (<= a_2 a_3))
  (=> (= pos_6 P4) (and (or (= pos_1 P1) (= pos_1 P3)) (not dir_1)))
  (=> (and (= pos_6 P4) (= pos_1 P1)) (<= c_1 a_2))
  (=> (and (= pos_6 P4) (= pos_1 P3)) (<= a_2 c_1))
  (=> (= pos_7 P1) (and (= pos_2 (ite dir_7 P4 P0)) (iff dir_2 dir_7)))
  (=> (= pos_7 P2) (and (= pos_2 (ite dir_7 P3 P0)) (iff dir_2 dir_7)))
  (=> (= pos_7 P3) (and (= pos_2 (ite dir_7 P4 P2)) (iff dir_2 dir_7)))
  (=> (= pos_7 P0) (and (or (= pos_2 P1) (= pos_2 P2)) dir_2))
  (=> (and (= pos_7 P0) (= pos_2 P1)) (<= a_3 a_2))
  (=> (and (= pos_7 P0) (= pos_2 P2)) (<= a_2 a_3))
  (=> (= pos_7 P4) (and (or (= pos_2 P1) (= pos_2 P3)) (not dir_2)))
  (=> (and (= pos_7 P4) (= pos_2 P1)) (<= c_1 a_2))
  (=> (and (= pos_7 P4) (= pos_2 P3)) (<= a_2 c_1))
  (=> (= pos_8 P1) (and (= pos_3 (ite dir_8 P4 P0)) (iff dir_3 dir_8)))
  (=> (= pos_8 P2) (and (= pos_3 (ite dir_8 P3 P0)) (iff dir_3 dir_8)))
  (=> (= pos_8 P3) (and (= pos_3 (ite dir_8 P4 P2)) (iff dir_3 dir_8)))
  (=> (= pos_8 P0) (and (or (= pos_3 P1) (= pos_3 P2)) dir_3))
  (=> (and (= pos_8 P0) (= pos_3 P1)) (<= a_3 a_2))
  (=> (and (= pos_8 P0) (= pos_3 P2)) (<= a_2 a_3))
  (=> (= pos_8 P4) (and (or (= pos_3 P1) (= pos_3 P3)) (not dir_3)))
  (=> (and (= pos_8 P4) (= pos_3 P1)) (<= c_1 a_2))
  (=> (and (= pos_8 P4) (= pos_3 P3)) (<= a_2 c_1))
  (= s1
   (+ a (ite (= pos P1) 1 0) (ite (= pos_2 P1) 1 0) (ite (= pos_1 P1) 1 0)
    (ite (= pos_3 P1) 1 0) (ite (= pos_4 P1) 1 0)))
  (= s3
   (+ c (ite (= pos P3) 1 0) (ite (= pos_2 P3) 1 0) (ite (= pos_1 P3) 1 0)
    (ite (= pos_3 P3) 1 0) (ite (= pos_4 P3) 1 0)))
  (= s2
   (+ a_1 (ite (= pos P2) 1 0) (ite (= pos_2 P2) 1 0) (ite (= pos_1 P2) 1 0)
    (ite (= pos_3 P2) 1 0) (ite (= pos_4 P2) 1 0)))
  (=> (= pos P1) (and (= a0 (ite dir P4 P0)) (iff d0 dir)))
  (=> (= pos P2) (and (= a0 (ite dir P3 P0)) (iff d0 dir)))
  (=> (= pos P3) (and (= a0 (ite dir P4 P2)) (iff d0 dir)))
  (=> (= pos P0) (and (or (= a0 P1) (= a0 P2)) d0))
  (=> (and (= pos P0) (= a0 P1)) (<= a_1 a))
  (=> (and (= pos P0) (= a0 P2)) (<= a a_1))
  (=> (= pos P4) (and (or (= a0 P1) (= a0 P3)) (not d0)))
  (=> (and (= pos P4) (= a0 P1)) (<= c a))
  (=> (and (= pos P4) (= a0 P3)) (<= a c))
  (=> (= pos_4 P1) (and (= a4 (ite dir_4 P4 P0)) (iff d4 dir_4)))
  (=> (= pos_4 P2) (and (= a4 (ite dir_4 P3 P0)) (iff d4 dir_4)))
  (=> (= pos_4 P3) (and (= a4 (ite dir_4 P4 P2)) (iff d4 dir_4)))
  (=> (= pos_4 P0) (and (or (= a4 P1) (= a4 P2)) d4))
  (=> (and (= pos_4 P0) (= a4 P1)) (<= a_1 a))
  (=> (and (= pos_4 P0) (= a4 P2)) (<= a a_1))
  (=> (= pos_4 P4) (and (or (= a4 P1) (= a4 P3)) (not d4)))
  (=> (and (= pos_4 P4) (= a4 P1)) (<= c a))
  (=> (and (= pos_4 P4) (= a4 P3)) (<= a c))
  (=> (= pos_1 P1) (and (= a2 (ite dir_1 P4 P0)) (iff d2 dir_1)))
  (=> (= pos_1 P2) (and (= a2 (ite dir_1 P3 P0)) (iff d2 dir_1)))
  (=> (= pos_1 P3) (and (= a2 (ite dir_1 P4 P2)) (iff d2 dir_1)))
  (=> (= pos_1 P0) (and (or (= a2 P1) (= a2 P2)) d2))
  (=> (and (= pos_1 P0) (= a2 P1)) (<= a_1 a))
  (=> (and (= pos_1 P0) (= a2 P2)) (<= a a_1))
  (=> (= pos_1 P4) (and (or (= a2 P1) (= a2 P3)) (not d2)))
  (=> (and (= pos_1 P4) (= a2 P1)) (<= c a))
  (=> (and (= pos_1 P4) (= a2 P3)) (<= a c))
  (=> (= pos_2 P1) (and (= a1 (ite dir_2 P4 P0)) (iff d1 dir_2)))
  (=> (= pos_2 P2) (and (= a1 (ite dir_2 P3 P0)) (iff d1 dir_2)))
  (=> (= pos_2 P3) (and (= a1 (ite dir_2 P4 P2)) (iff d1 dir_2)))
  (=> (= pos_2 P0) (and (or (= a1 P1) (= a1 P2)) d1))
  (=> (and (= pos_2 P0) (= a1 P1)) (<= a_1 a))
  (=> (and (= pos_2 P0) (= a1 P2)) (<= a a_1))
  (=> (= pos_2 P4) (and (or (= a1 P1) (= a1 P3)) (not d1)))
  (=> (and (= pos_2 P4) (= a1 P1)) (<= c a))
  (=> (and (= pos_2 P4) (= a1 P3)) (<= a c))
  (=> (= pos_3 P1) (and (= a3 (ite dir_3 P4 P0)) (iff d3 dir_3)))
  (=> (= pos_3 P2) (and (= a3 (ite dir_3 P3 P0)) (iff d3 dir_3)))
  (=> (= pos_3 P3) (and (= a3 (ite dir_3 P4 P2)) (iff d3 dir_3)))
  (=> (= pos_3 P0) (and (or (= a3 P1) (= a3 P2)) d3))
  (=> (and (= pos_3 P0) (= a3 P1)) (<= a_1 a))
  (=> (and (= pos_3 P0) (= a3 P2)) (<= a a_1))
  (=> (= pos_3 P4) (and (or (= a3 P1) (= a3 P3)) (not d3)))
  (=> (and (= pos_3 P4) (= a3 P1)) (<= c a))
  (=> (and (= pos_3 P4) (= a3 P3)) (<= a c)) (<= 2 s1) (<= (+ s2 3) s3)))
(check-sat)
(get-value
 (a a_1 a_2 a_3 a_4 a_5 a_6 a_7 a0 a1 a2 a3 a4 c c_1 c_2 c_3 d0 d1 d2 d3 d4
  dir dir_1 dir_2 dir_3 dir_4 dir_5 dir_6 dir_7 dir_8 dir_9 dir_10 dir_11
  dir_12 dir_13 dir_14 dir_15 dir_16 dir_17 dir_18 dir_19 pos pos_1 pos_2
  pos_3 pos_4 pos_5 pos_6 pos_7 pos_8 pos_9 pos_10 pos_11 pos_12 pos_13
  pos_14 pos_15 pos_16 pos_17 pos_18 pos_19 s1 s2 s3))
