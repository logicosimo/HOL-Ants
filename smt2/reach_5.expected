sat
((a 0)
 (a_1 10)
 (a_2 0)
 (a_3 10)
 (a_4 0)
 (a_5 10)
 (a_6 0)
 (a_7 0)
 (a0 P2)
 (a1 P2)
 (a2 P2)
 (a3 P2)
 (a4 P2)
 (a5 P2)
 (a6 P2)
 (a7 P2)
 (a8 P2)
 (a9 P2)
 (c 10)
 (c_1 10)
 (c_2 0)
 (c_3 0)
 (d0 false)
 (d1 false)
 (d2 false)
 (d3 false)
 (d4 false)
 (d5 false)
 (d6 false)
 (d7 false)
 (d8 false)
 (d9 false)
 (dir false)
 (dir_1 false)
 (dir_2 false)
 (dir_3 false)
 (dir_4 false)
 (dir_5 false)
 (dir_6 false)
 (dir_7 false)
 (dir_8 false)
 (dir_9 false)
 (dir_10 true)
 (dir_11 true)
 (dir_12 true)
 (dir_13 true)
 (dir_14 true)
 (dir_15 true)
 (dir_16 true)
 (dir_17 true)
 (dir_18 true)
 (dir_19 true)
 (dir_20 true)
 (dir_21 true)
 (dir_22 true)
 (dir_23 true)
 (dir_24 true)
 (dir_25 true)
 (dir_26 true)
 (dir_27 true)
 (dir_28 true)
 (dir_29 true)
 (dir_30 true)
 (dir_31 true)
 (dir_32 true)
 (dir_33 true)
 (dir_34 true)
 (dir_35 true)
 (dir_36 true)
 (dir_37 true)
 (dir_38 true)
 (dir_39 true)
 (pos P3)
 (pos_1 P3)
 (pos_2 P3)
 (pos_3 P3)
 (pos_4 P3)
 (pos_5 P3)
 (pos_6 P3)
 (pos_7 P3)
 (pos_8 P3)
 (pos_9 P3)
 (pos_10 P4)
 (pos_11 P4)
 (pos_12 P4)
 (pos_13 P4)
 (pos_14 P4)
 (pos_15 P4)
 (pos_16 P4)
 (pos_17 P4)
 (pos_18 P4)
 (pos_19 P4)
 (pos_20 P3)
 (pos_21 P3)
 (pos_22 P3)
 (pos_23 P3)
 (pos_24 P3)
 (pos_25 P3)
 (pos_26 P3)
 (pos_27 P3)
 (pos_28 P3)
 (pos_29 P3)
 (pos_30 P2)
 (pos_31 P2)
 (pos_32 P2)
 (pos_33 P2)
 (pos_34 P2)
 (pos_35 P2)
 (pos_36 P2)
 (pos_37 P2)
 (pos_38 P2)
 (pos_39 P2)
 (s1 0)
 (s2 10)
 (s3 20))
